File: java/gameplay-db-adapter/src/main/java/com/hexagonalchess/gameplay/InMemoryMoveRepository.java
package com.hexagonalchess.gameplay;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class InMemoryMoveRepository implements MoveRepository {

    private final Map<BoardId, List<Move>> store = new HashMap<>();

    @Override
    public void save(Move move, BoardId boardId) {
        store.computeIfAbsent(boardId, k -> new ArrayList<>()).add(move);
    }

    @Override
    public List<Move> findByBoardId(BoardId boardId) {
        return store.getOrDefault(boardId, List.of());
    }
}

File: ruby/gameplay_db_adapter/lib/gameplay/in_memory_move_repository.rb
require 'gameplay/move_repository'

module Gameplay
  class InMemoryMoveRepository
    include MoveRepository

    def initialize
      @store = Hash.new { |h, k| h[k] = [] }
    end

    def save(move, board_id)
      @store[board_id] << move
    end

    def find_by_board_id(board_id)
      @store[board_id]
    end
  end
end
